### ISSUE TITLE
eBPF: Cannot load program rnetdata_ebpf_process.o

### ISSUE THREAD

**Issue by fayak** [netdata/netdata#8164]
##### Bug report summary

I'm trying to test the new eBPF functionnality of netdata. I followed the tutorial https://docs.netdata.cloud/collectors/ebpf_process.plugin/, but when I start netdata, there's no ebpf on the dashboard, and the error logs contains:
```
$ cat /var/log/netdata/error.log | grep ebpf
2020-02-23 01:00:05: netdata INFO  : PLUGINSD[ebpf_process] : thread created with task id 14586
2020-02-23 01:00:05: netdata INFO  : PLUGINSD[ebpf_process] : set name of thread 14586 to PLUGINSD[ebpf_p
2020-02-23 01:00:05: netdata INFO  : PLUGINSD[ebpf_process] : connected to '/usr/libexec/netdata/plugins.d/ebpf_process.plugin' running on pid 14588
2020-02-23 01:00:05: ebpf_process.plugin ERROR : MAIN : [EBPF_PROCESS] Cannot load program: /usr/libexec/netdata/plugins.d/rnetdata_ebpf_process.o (errno 22, Invalid argument)
2020-02-23 01:00:05: netdata ERROR : PLUGINSD[ebpf_process] : read failed: end of file
2020-02-23 01:00:05: netdata ERROR : PLUGINSD[ebpf_process] : '/usr/libexec/netdata/plugins.d/ebpf_process.plugin' (pid 14588) disconnected after 0 successful data collections (ENDs).
2020-02-23 01:00:05: netdata INFO  : PLUGINSD[ebpf_process] : '/usr/libexec/netdata/plugins.d/ebpf_process.plugin' (pid 14588) does not generate useful output but it reports success (exits with 0). Waiting a bit before starting it again..
2020-02-23 01:00:15: netdata INFO  : PLUGINSD[ebpf_process] : connected to '/usr/libexec/netdata/plugins.d/ebpf_process.plugin' running on pid 15167
2020-02-23 01:00:15: ebpf_process.plugin ERROR : MAIN : [EBPF_PROCESS] Cannot load program: /usr/libexec/netdata/plugins.d/rnetdata_ebpf_process.o (errno 22, Invalid argument)
2020-02-23 01:00:15: netdata ERROR : PLUGINSD[ebpf_process] : read failed: end of file
2020-02-23 01:00:15: netdata ERROR : PLUGINSD[ebpf_process] : '/usr/libexec/netdata/plugins.d/ebpf_process.plugin' (pid 15167) disconnected after 0 successful data collections (ENDs).
2020-02-23 01:00:15: netdata INFO  : PLUGINSD[ebpf_process] : '/usr/libexec/netdata/plugins.d/ebpf_process.plugin' (pid 15167) does not generate useful output but it reports success (exits with 0). Waiting a bit before starting it again..
2020-02-23 01:00:25: ebpf_process.plugin ERROR : MAIN : [EBPF_PROCESS] Cannot load program: /usr/libexec/netdata/plugins.d/rnetdata_ebpf_process.o (errno 22, Invalid argument)
2020-02-23 01:00:35: ebpf_process.plugin ERROR : MAIN : [EBPF_PROCESS] Cannot load program: /usr/libexec/netdata/plugins.d/rnetdata_ebpf_process.o (errno 22, Invalid argument)
2020-02-23 01:00:45: ebpf_process.plugin ERROR : MAIN : [EBPF_PROCESS] Cannot load program: /usr/libexec/netdata/plugins.d/rnetdata_ebpf_process.o (errno 22, Invalid argument)
2020-02-23 01:00:55: ebpf_process.plugin ERROR : MAIN : [EBPF_PROCESS] Cannot load program: /usr/libexec/netdata/plugins.d/rnetdata_ebpf_process.o (errno 22, Invalid argument)
2020-02-23 01:01:05: ebpf_process.plugin ERROR : MAIN : [EBPF_PROCESS] Cannot load program: /usr/libexec/netdata/plugins.d/rnetdata_ebpf_process.o (errno 22, Invalid argument)
2020-02-23 01:01:15: ebpf_process.plugin ERROR : MAIN : [EBPF_PROCESS] Cannot load program: /usr/libexec/netdata/plugins.d/rnetdata_ebpf_process.o (errno 22, Invalid argument)
2020-02-23 01:01:25: ebpf_process.plugin ERROR : MAIN : [EBPF_PROCESS] Cannot load program: /usr/libexec/netdata/plugins.d/rnetdata_ebpf_process.o (errno 22, Invalid argument)
2020-02-23 01:01:35: ebpf_process.plugin ERROR : MAIN : [EBPF_PROCESS] Cannot load program: /usr/libexec/netdata/plugins.d/rnetdata_ebpf_process.o (errno 22, Invalid argument)
2020-02-23 01:01:45: ebpf_process.plugin ERROR : MAIN : [EBPF_PROCESS] Cannot load program: /usr/libexec/netdata/plugins.d/rnetdata_ebpf_process.o (errno 22, Invalid argument)
```

```
# file /usr/libexec/netdata/plugins.d/rnetdata_ebpf_process.o
/usr/libexec/netdata/plugins.d/rnetdata_ebpf_process.o: ELF 64-bit LSB relocatable, *unknown arch 0xf7* version 1 (SYSV), not stripped
```

##### OS / Environment

I'm running debian 10, all things are up-to-date

```
Linux ... 4.19.0-8-amd64 #1 SMP Debian 4.19.98-1 (2020-01-26) x86_64 GNU/Linux
```

##### Netdata version
```
netdata -V
netdata v1.20.0-7-nightly
```

##### Component Name

ebpf_process.plugin I believe

##### Steps To Reproduce

The steps given in https://docs.netdata.cloud/collectors/ebpf_process.plugin/

##### Expected behavior

Having ebpf charts

Thanks for the help !

**Comment by fayak** [comment]
I just tried, it does work on archlinux

**Comment by thiagoftsm** [comment]
Hi @fayak ,

Firstly we apologize for the problem!
You had problem with Debian, because our CI is currently working only with the latest version of the kernel 4.19 to generate the eBPF programs, but it looks like that after the update from 4.19.102 to 4.19.103 we cannot compile more one unique eBPF program for the whole 4.19 release.
We are working to extend our CI and deliver eBPF programs for all available kernels.

Thank you to give us your feedback.

Best regards!

**Comment by prologic** [comment]
@fayak Out of interest what is the Kernel version on Arch you tested with that you report is working?

**Comment by fayak** [comment]
@thiagoftsm no need to apologize ! Thanks for your quick answer and the work put in the Netdata project !

@prologic `Linux ... 5.5.4-arch1-1 #1 SMP PREEMPT Sat, 15 Feb 2020 00:36:29 +0000 x86_64 GNU/Linux` I believe this was the last one available yesterday, since the 5.5.5 just came out today on the official repositories (I haven't tested it yet)

**Comment by thiagoftsm** [comment]
All right @fayak !
An additional information, for the family 5.x there was not yet a big change in the eBPF that force us to create two different eBPF programs for it, so the program used on 5.0 is the same used for 5.5, but I am seeing that we are having big changes coming with 5.6, probably we will need to compile a new exclusively for it.

**Comment by prologic** [comment]
Thanks @fayak This all makes sense. We're busily working on getting a new variant out for the 4.15 series as quickly as we can!

**Comment by prologic** [comment]
@thiagoftsm So do we need a specific `4.19.103` variant here?

**Comment by prologic** [comment]
Ignore my last comment; We either need a `4.19` variant or `4.19.98` I _thik_

**Comment by thiagoftsm** [comment]
Yes, we will need two variants, one that worked until the kernel `4.19.10x` and other that is working the latest kernels `4.19`, during this week I will verify the correct version that breaks the compatibility among the kernel `4.19.x`, because I remember I compiled on all `4.19.9x`  without any problems in a near past, but it cannot work now.

**Comment by thiagoftsm** [comment]
Hi @fayak ,

We merged the solution for the kernel 4.19 used by Debian. During my tests I used the Debian 10.2 to confirm that we could run the collector.

Best regards!
